tests/join_test.py

```diff
@@ -105,6 +105,27 @@ def test_inner_join(self):
         with self.assertRaises(ValueError):
             l, r = ak.join.inner_join(left, right, wherefunc=ak.intersect1d, whereargs=(ak.arange(10), ak.arange(5)))
 
+    def test_lookup(self):
+        keys = ak.arange(5)
+        values = 10*keys
+        args = ak.array([5, 3, 1, 4, 2, 3, 1, 0])
+        ans = np.array([-1, 30, 10, 40, 20, 30, 10, 0])
+        # Simple lookup with int keys
+        # Also test shortcut for unique-ordered keys
+        res = ak.lookup(keys, values, args, fillvalue=-1, keys_from_unique=True)
+        self.assertTrue((res.to_ndarray() == ans).all())
```

Comment:
I know it's not consistent across our tests but i prefer
```python
self.assertListEqual(res.to_ndarray().tolist(), ans.tolist())
```
because it gives more precise information about where the lists are different

for example
```python
>       self.assertTrue((ak.array([1,2,3]) == ak.array([1,1,3])).all())
E       AssertionError: False is not true
```
vs
```python
>       self.assertListEqual(ak.array([1,2,3]).to_ndarray().tolist(), ak.array([1,1,3]).to_ndarray().tolist())
E       AssertionError: Lists differ: [1, 2, 3] != [1, 1, 3]
E       
E       First differing element 1:
E       2
E       1
E       
E       - [1, 2, 3]
E       ?     ^
E       
E       + [1, 1, 3]
E       ?     ^
```